dts/bindings/video/zephyr,sw-generator.yaml

```diff
@@ -0,0 +1,11 @@
+# Copyright 2024 tinyVision.ai Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+compatible: "zephyr,sw-generator"
```

Comment:
why does this need to be in DT?

Reply:
I personally needed it for exposing the hue/saturation/brightness controls over USB Video.
https://github.com/zephyrproject-rtos/zephyr/pull/76798/files#diff-47c7b3f77fda7a7bc55ec72fb80259f4e7b7111fd1f20e0653f28ea90d663b76R13

This permit to refer to this "virtual hardware" as a target for video-controls, using a `phandle` and avoid a special-case in this driver, or in the various samples.

It sounds a bit of a deformation of the purpose of devicetree describing actual hardware.
Having it on the devicetree was simply a method to avoid making a special case of it everywhere

I'd be glad to search an alternative method if it sounds more suited!

Reply:
I have the same question. It seems still unclear why we need to have this in DT. The video capture sample currently fallback by default to video sw generator if we don't specify any shield.